cfme/tests/cloud/test_quota.py

```diff
@@ -0,0 +1,326 @@
+            "environment": {"automatic_placement": True},
+            "properties": {"instance_type": partial_match(instance_type)},
+            "customize": {
+                "admin_username": provisioning["customize_username"],
+                "root_password": provisioning["customize_password"],
+            },
+        }
+
+
+@pytest.fixture
+def set_child_tenant_quota(request, appliance, new_child):
+    """This fixture assigns quota to child tenant"""
+    field, value = request.param
+    new_child.set_quota(**{"{}_cb".format(field): True, field: value})
+    yield
+    # will refresh page as navigation to configuration is blocked if alerts are on the page
+    appliance.server.login_admin()
+    appliance.server.browser.refresh()
+    new_child.set_quota(**{"{}_cb".format(field): False})
+
+
+@pytest.fixture
+def set_project_quota(request, appliance, new_project):
+    """This fixture assigns quota to project"""
+    field, value = request.param
+    new_project.set_quota(**{"{}_cb".format(field): True, field: value})
+    yield
+    # will refresh page as navigation to configuration is blocked if alerts are on the page
+    appliance.server.login_admin()
+    appliance.server.browser.refresh()
+    new_project.set_quota(**{"{}_cb".format(field): False})
+
+
+@pytest.fixture(scope="module")
+def new_tenant(appliance):
+    """This fixture creates new tenant under root tenant(My Company)"""
+    collection = appliance.collections.tenants
+    tenant = collection.create(
+        name="tenant_{}".format(fauxfactory.gen_alphanumeric()),
+        description="tenant_des{}".format(fauxfactory.gen_alphanumeric()),
+        parent=collection.get_root_tenant(),
+    )
+    yield tenant
+    if tenant.exists:
+        tenant.delete()
+
+
+@pytest.fixture(scope="module")
+def new_child(appliance, new_tenant):
+    """The fixture creates new child tenant"""
+    collection = appliance.collections.tenants
+    child_tenant = collection.create(
+        name="tenant_{}".format(fauxfactory.gen_alphanumeric()),
+        description="tenant_des{}".format(fauxfactory.gen_alphanumeric()),
+        parent=new_tenant,
+    )
+    yield child_tenant
+    if child_tenant.exists:
+        child_tenant.delete()
+
+
+@pytest.fixture(scope="module")
+def new_group_child(appliance, new_child, new_tenant):
+    """This fixture creates new group assigned by new child tenant"""
+    collection = appliance.collections.groups
+    group = collection.create(
+        description="group_{}".format(fauxfactory.gen_alphanumeric()),
+        role="EvmRole-super_administrator",
+        tenant="My Company/{parent}/{child}".format(parent=new_tenant.name, child=new_child.name),
+    )
+    yield group
+    if group.exists:
+        group.delete()
+
+
+@pytest.fixture(scope="module")
+def new_user_child(appliance, new_group_child):
+    """This fixture creates new user which assigned to new child tenant"""
+    collection = appliance.collections.users
```

Comment:
@ganeshhubale good practice is when we need one thing multiple time then only we use assigned like this. but the collection is required one time here. then it's not a practice.